File: javabackend/src/main/java/mft/dev/javabackend/model/Todo.java
package mft.dev.javabackend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "todos")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Todo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    @Column(name = "titolo")
    private String titolo;

    @Column(name = "descrizione")
    private String descrizione;

    @ManyToOne
    @JoinColumn(name = "utente_id")
    private Utente utente;
}

File: kotlinbackend/src/main/kotlin/mft/dev/kotlinbackend/model/Todo.kt
package mft.dev.kotlinbackend.model
import jakarta.persistence.*

@Entity
@Table(name = "todos")
data class Todo(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    @Column(name = "titolo")
    var titolo: String,
    @Column(name = "descrizione")
    var descrizione: String?,
    @ManyToOne
    @JoinColumn(name = "utente_id")
    var utente: Utente? = null
)
